src/main/scala/tile/Interrupts.scala

```diff
@@ -50,7 +50,7 @@ trait HasExternalInterrupts { this: BaseTile =>
 
   // TODO: the order of the following two functions must match, and
   //         also match the order which things are connected to the
-  //         per-tile crossbar in subsystem.HasRocketTiles
+  //         per-tile crossbar in subsystem.HasTiles.connectInterrupts
```

Comment:
Will this code actually still work right if optional Debugs, CLINTs or PLICs are not present?

Reply:
Only for the ones that are currently optional. This is a stab at the API; I didn’t fix the fundamental problems.